src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,13 +271,60 @@ def algorithm(self, parameters=None):
             with CodeTimer("Rucio block resolution", logger=logging):
                 self.findRucioBlocks(uniqueActivePU, msPileupList)
 
-            # TODO in future tickets: find the json files in the spec/sandbox area and tweak them
-
+            # loop over existing pileup names and
+            # find the json files in the spec/sandbox area and tweak them
+            adjustJSONSpec(msPileupList)
         except Exception as ex:
             msg = f"Caught unexpected exception in WorkflowUpdater. Details:\n{str(ex)}"
             logging.exception(msg)
             raise WorkflowUpdaterException(msg) from None
 
+    def adjustJSONSpec(self, msPileupList):
+        """
+        Main logic of the algorithm:
+        - for every pileup record find out location of tarball
+        - get configuration files within tarball
+        - extract block information
+        - replace pileupconf.json files within tarball
+        :param msPileupList: input list of pileup records
+        :return: nothing, it perform checks and adjust in place JSON spec file(s)
+        """
+        # loop over existing pileup names and
+        # find the json files in the spec/sandbox area and tweak them
+        for rec in msPileupList:
+            # find out location of pileup tarball
+            files = findFiles(self.idir, rec['pileupName'])
+            if len(files) != 1:
+                self.logger.warning("For pilupe %s found %s", rec['pileupName'], files)
+                continue
+            # name of sandbox tar file
+            tfile = files[0]
+
+            # get list of pileupconf.json files within our tar ball file
+            pileupConfFiles = findJsonSandboxFiles(tfile)
+
+            # extract blocks from msPileupList structures
+            blocks = rec['blocks']
+#                 blocks = [b for d in msPileupList for b in d["blocks"]]
+
+            # construct jdict={"path-to-json1": jdoc1, "path-to-json2": jdoc2}
+            jdict = {}
+            for fname in pileupConfFiles:
```

Comment:
if there are multiple files using the same pileup, we should not redo all these operations. Instead, we reuse the result that we found for the first pileup configuration.
In other words, a workflow can have multiple steps/tasks, and some of those steps/tasks might be using the very same pileup. Looking into the workflow sandbox, you will see that there is a new pileupconf.json inside each step (cmsRunXXX) or task directory. So, we should make sure we update those efficiently, doing all the hard work only once, and simply copying it for the remainders.

Reply:
Yes, I agree with this one. Can you please add a comment with TODO FIXME to address this in the future?

Reply:
I put this comment in a code directly see https://github.com/dmwm/WMCore/pull/11884/commits/ace826a9f59d8b7fd42c5eee19938ce65b82b859#diff-e42968b3836140494b36bc2f6a339ba46c737f8e6de3117c8f81e5cec97320a9R318

Reply:
This is resolved.